lib/core/src/recover/recoverer.rs

```diff
@@ -70,8 +70,13 @@ impl Recoverer {
 
         let mut preimages = HashMap::new();
         for (swap_id, claim_tx) in claim_txs_by_swap_id {
-            if let Ok(preimage) = Self::get_send_swap_preimage_from_claim_tx(swap_id, &claim_tx) {
-                preimages.insert(swap_id, preimage);
+            match Self::get_send_swap_preimage_from_claim_tx(swap_id, &claim_tx) {
+                Ok(preimage) => {
+                    preimages.insert(swap_id.to_string(), preimage);
+                }
+                Err(e) => {
+                    error!("Failed to get swap preimage from claim tx for swap {swap_id}: {e}");
```

Comment:
Can you log the tx_id also so we can check it?

Reply:
[7e5b0b2](https://github.com/breez/breez-sdk-liquid/pull/653/commits/7e5b0b2b901f469b3ba52579fa9b0a1b2dd85467)